x-pack/plugins/fleet/server/services/epm/packages/remove.ts

```diff
@@ -237,9 +266,9 @@ async function deleteAssets(
     // then the other asset types
     await Promise.all([
       ...deleteESAssets(otherAssets, esClient),
-      deleteKibanaAssets(installedKibana, spaceId),
+      resolveAndDeleteKibanaAssets(installedKibana, spaceId),
```

Comment:
Probably we can add the check here too. Where is this code path used? I'm not super familiar with this code, so I'd rely on your opinion.

Reply:
It is being called when deleting a package or when rolling back a package installation. I think it would be good to add the same version check here too.

Reply:
Okay, I'll be introducing another optimization in the remove method soon. I'll address your suggestion in that PR.